vyper/builtin_functions/functions.py

```diff
@@ -97,27 +102,43 @@
     vyper_warn,
 )
 
-from .signatures import Optional, validate_inputs
+from .signatures import TypeTypeDefinition, validate_inputs
 
 SHA256_ADDRESS = 2
 SHA256_BASE_GAS = 60
 SHA256_PER_WORD_GAS = 12
 
 
-class _BuiltinFunction:
-    def __repr__(self):
-        return f"builtin function {self._id}"
+class _SimpleBuiltinFunction:
 
+    _kwargs: Dict[str, KwargSettings] = {}
 
-class _SimpleBuiltinFunction(_BuiltinFunction):
-    def fetch_call_return(self, node):
+    def _validate_arg_types(self, node):
         validate_call_args(node, len(self._inputs), getattr(self, "_kwargs", []))
+
         for arg, (_, expected) in zip(node.args, self._inputs):
             validate_expected_type(arg, expected)
 
+        return
+
+    def fetch_call_return(self, node):
+        # Call infer_arg_types to validate arguments and derive the type definitions
+        # for annotation
```

Comment:
this can be removed now since it's pretty clear what `_validate_arg_types()` does
```suggestion
```